internal/sync/lifecycle.go

```diff
@@ -0,0 +1,32 @@
+package sync
+
+// LifecycleOnce is an abstraction around a lifecycle interface
```

Comment:
It's not "around" it. It makes it easier to implement a Lifecycle type with
correct behavior.